RedHatInsights/drift-frontend
# [DRFT-275] Handle no hsp read permissions

This PR blocks the HSP icon on:
Add system modal
Comparison table
Create baseline modal

To test:
- Go to this file: https://github.com/johnsonm325/drift-frontend/blob/9cadc3152ab41c4f7c3281606409af07b1a3ce76/src/App.js#L31-L40 and change to this:
```
    const handlePermissionsUpdate = (hasCompareRead, hasBaselinesRead, hasBaselinesWrite/*, hasHSPRead*/, hasInventoryRead) => {
        setPermissions({
            hasCompareReadPermissions: hasCompareRead,
            hasBaselinesReadPermissions: hasBaselinesRead,
            hasBaselinesWritePermissions: hasBaselinesWrite,
            hasHSPReadPermissions: /*hasHSPRead*/false,
            hasInventoryReadPermissions: hasInventoryRead,
            arePermissionsLoaded: true
        });
    };
```
- For each of the following, there should be no historical profiles column:
    - Go to inventory/systems table on add system modal
        - Drift -> Comparison -> click 'Add to comparison' button -> make sure 'Systems' tab is selected
    - Go to inventory/systems table on create baseline modal
        - Drift -> Baselines -> click 'Create baseline' button -> click 'Copy an existing system'
- If you create a comparison that includes a system with hsps or an hsp by itself, there should be no hsp icon in the header of the table under those items.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
